lua/luasnip/nodes/insertNode.lua

```diff
@@ -37,8 +66,9 @@ local function I(pos, static_text, opts)
 			type = types.insertNode,
 			inner_active = false,
 			input_active = false,
-		}, opts)
+		}, node_opts)
 	end
+	---@cast node LuaSnip.InsertNode|LuaSnip.ExitNode
```

Comment:
Does luals understand what's going on if you put this into the respective branches of the `if`?

Reply:
probably yes, you'd prefer to do this?